### PR TITLE
Billing: big refactor and design cleanup

### PR DESCRIPTION
**Pull request by mtias**
The Billing History page is really old, still relies on `overlay` component for receipts and has a ton of custom code and CSS.

This PR cleans the majority of these issues, removes the overlay, and reuses core UI components instead.

Before:
image

After:
image

### CODE REVIEW

**Comment by rickybanister** [comment]
Wow, that looks great.

One thought, don't our form buttons float right with the primary button all the way to the right?

**Comment by mtias** [comment]
Yes, still more stuff to clean up :)

**Comment by mtias** [comment]
@rickybanister updated.

**Comment by gziolo** [comment]
@mtias please ping me when you are done, I'm happy to review your changes ;)

**Comment by mtias** [comment]
@gziolo calling it done for a first pass, feel free to go through it :)

**Comment by gziolo** [comment]
I'll do as first thing tomorrow morning, just finished for today :)

File: client/me/billing-history/billing-history-table.jsx [hunk 1]
```diff
@@ -77,8 +77,8 @@ const BillingHistoryTable = React.createClass( {
 
 	renderTransaction: function( transaction ) {
 		return (
-			<div className="transaction-links">
-				<a className="view-receipt" href={ '/me/billing/' + transaction.id } onClick={ this.recordClickEvent( 'View Receipt in Billing History' ) } >
+			<div className="billing-history__transaction-links">
+				<a className="billing-history__view-receipt" href={ '/me/billing/' + transaction.id } onClick={ this.recordClickEvent( 'View Receipt in Billing History' ) } >
```
Comment (gwwar) [inline client/me/billing-history/billing-history-table.jsx]:
```javascript
/me/billing/${ transaction.id }
```

File: client/me/billing-history/main.jsx [hunk 2]
```diff
@@ -2,7 +2,6 @@
  * External dependencies
  */
 var React = require( 'react' ),
```
Comment (gwwar) [inline client/me/billing-history/main.jsx]: Let's es6ify the rest of the imports.

File: client/me/billing-history/style.scss [hunk 3]
```diff
@@ -0,0 +1,454 @@
...
+	position: absolute;
+	top: 35px;
+	z-index: z-index( 'root', '.billing-history-page .filter-popover-content' );
+	border-radius: 4px;
+	background-color: $white;
+	box-shadow: 0 1px 6px rgba(0,0,0,0.2), 0 0 25px 10px rgba(0,0,0,0.1);
+	outline: 0;
+
+	&.datepicker {
+		right: -115px;
```
Comment (gwwar) [inline client/me/billing-history/style.scss]: These numbers seem somewhat magical.

File: client/me/billing-history/style.scss [hunk 4]
```diff
@@ -0,0 +1,454 @@
...
+}
+.filter-popover-content td.descriptor {
+	font-weight: 400;
+	font-size: 13px;
+}
+.filter-popover-content td.transactions-header__count {
+	font-weight: 600;
+	font-size: 14px;
+	text-align: right;
+	color: #777;
```
Comment (gwwar) [inline client/me/billing-history/style.scss]: Forgot to remove this? This will be overwritten by the line below

**Comment by gwwar** [comment]
Code is looking good so far but it looks like the billing history section was deleted, but the import was left in style.scss
```javascript
@import 'sections/billing-history';            // Billing History styles
```
Ping me again when you've updated.

File: client/me/billing-history/main.jsx (hunk 2 above)
Comment (mtias) [inline client/me/billing-history/main.jsx]: I was going to do this in a separate PR to make ti easier to review and spot issues in this one.

File: client/me/billing-history/style.scss (hunk 3 above)
Comment (mtias) [inline client/me/billing-history/style.scss]: We need to get rid of the whole `filter-popover` thing and use `<Popover>`.

File: client/me/billing-history/controller.js [hunk 5]
```diff
@@ -37,7 +37,7 @@ export default {
 
 			renderWithReduxStore(
```
Comment (gziolo) [inline client/me/billing-history/controller.js]: We render twice into `primary` element, when there is a transaction id provided.

**Comment by gziolo** [comment]
I guess it's completely unrelated but we need to improve the following user experience when filtering transactions and no results are found:
screen shot 2016-04-13 at 12 20 32

**Comment by gziolo** [comment]
There are two warnings on JS console:
screen shot 2016-04-13 at 12 22 45

File: client/me/billing-history/view-receipt-modal.jsx [hunk 6]
```diff
@@ -11,27 +11,76 @@ var React = require( 'react' ),
...
 	observe = require( 'lib/mixins/data-observe' ),
-	Overlay = require( 'components/overlay' ),
 	eventRecorder = require( 'me/event-recorder' );
 
+import Card from 'components/card';
+import Main from 'components/main';
+import HeaderCake from 'components/header-cake';
+
 module.exports = React.createClass( {
 	displayName: 'ViewReceiptModal',
```
Comment (gziolo) [inline client/me/billing-history/view-receipt-modal.jsx]: It's minor, but it's no longer modal component :)

**Comment by folletto** [comment]
Suggestion to simplify that block and making it more aligned with our patterns:

screen shot 2016-04-13 at 13 08 00

**Comment by gwwar** [comment]
👍  Looking good, feel free to 🚢  after this case is cleared up. In small viewports your avatar will overlap with the header cake.

screen shot 2016-04-14 at 6 41 28 am

**Comment by mtias** [comment]
@gwwar I believe I fixed that in master yesterday, but this branch hasn't picked it up :)

**Comment by mtias** [comment]
Confirmed fixed with rebase: 
image

**Comment by gziolo** [comment]
Nice job! :+1:

File: client/me/billing-history/main.jsx [hunk 7]
```diff
@@ -18,63 +17,40 @@ var observe = require( 'lib/mixins/data-observe' ),
...
-							}
-						</p>
-					</div>
-
+				<Card className="billing-history__receipts">
 					<BillingHistoryTable transactions={ data.billingHistory } />
 				</Card>
-
+				<Card href={ puchasesPaths.list() }>
+					{ this.translate( 'Go to "Purchases" to add or cancel a plan.' ) }
```
Comment (scruffian) [inline client/me/billing-history/main.jsx]: Could you explain why we added this section? Is that because users miss the link in the header cake?
